pkg/satellite/satelliteweb/api.go

```diff
@@ -68,32 +77,30 @@ func getToken(req *http.Request) string {
 }
 
 // getQuery retrieves graphql query from request
-func getQuery(req *http.Request) (query string, err error) {
+func getQuery(req *http.Request) (query graphqlJSON, err error) {
 	switch req.Method {
 	case http.MethodGet:
-		return req.URL.Query().Get(satelliteql.Query), nil
+		query.Query = req.URL.Query().Get(satelliteql.Query)
+		return query, nil
 	case http.MethodPost:
-		return queryPOST(req)
+		return query, queryPOST(req, &query)
```

Comment:
From `queryPost` return `query` and the error, then you can do here `return queryPost(req)`